// bootstrap-vue-next/dist/src/components/index.mjs
import { _, a } from "../../BAccordionItem.vue_vue_type_script_setup_true_lang-CIm9Esfp.mjs";
import { B } from "../../BAlert-ChvVF808.mjs";
import { _ as _2, a as a2 } from "../../BAvatarGroup.vue_vue_type_script_setup_true_lang-DEn76BWm.mjs";
import { _ as _3 } from "../../BBadge.vue_vue_type_script_setup_true_lang-CYoUdtux.mjs";
import { _ as _4, a as a3 } from "../../BBreadcrumb.vue_vue_type_script_setup_true_lang-CGIh5Dyn.mjs";
import { _ as _5 } from "../../BButton.vue_vue_type_script_setup_true_lang-DUrERyIC.mjs";
import { _ as _6, a as a4 } from "../../BButtonToolbar.vue_vue_type_script_setup_true_lang-DeG8adnx.mjs";
import { _ as _7 } from "../../BCloseButton.vue_vue_type_script_setup_true_lang-B_GsCJs6.mjs";
import { _ as _8, a as a5, b, c, d, e, f } from "../../BCard.vue_vue_type_script_setup_true_lang-Bmyix7rF.mjs";
import { _ as _9, a as a6 } from "../../BCardText.vue_vue_type_script_setup_true_lang-DHmRlQE7.mjs";
import { _ as _10, a as a7 } from "../../BCarouselSlide.vue_vue_type_script_setup_true_lang-CGqYLLHu.mjs";
import { _ as _11 } from "../../BCollapse.vue_vue_type_script_setup_true_lang-DMQdKnW0.mjs";
import { _ as _12 } from "../../BCol.vue_vue_type_script_setup_true_lang-CpPd5cQ6.mjs";
import { _ as _13 } from "../../BContainer.vue_vue_type_script_setup_true_lang-CRsJ73Pp.mjs";
import { _ as _14 } from "../../BRow.vue_vue_type_script_setup_true_lang-1tXs9iie.mjs";
import { _ as _15 } from "../../BDropdown.vue_vue_type_script_setup_true_lang-CCBMSTuN.mjs";
import { _ as _16, a as a8, b as b2, c as c2, d as d2, e as e2, f as f2 } from "../../BDropdownText.vue_vue_type_script_setup_true_lang-ChLJg_WW.mjs";
import { _ as _17 } from "../../BForm.vue_vue_type_script_setup_true_lang-BniLBRHQ.mjs";
import { _ as _18, a as a9 } from "../../BFormFloatingLabel.vue_vue_type_script_setup_true_lang-B9qmgxL3.mjs";
import { _ as _19, a as a10, b as b3, c as c3 } from "../../BFormValidFeedback.vue_vue_type_script_setup_true_lang-DMdYTYKn.mjs";
import { _ as _20, a as a11 } from "../../BFormCheckboxGroup.vue_vue_type_script_setup_true_lang-BD7pZJij.mjs";
import { _ as _21 } from "../../BFormFile.vue_vue_type_script_setup_true_lang-BEtcRM_N.mjs";
import { _ as _22 } from "../../BFormGroup.vue_vue_type_script_lang-CfL_PZZX.mjs";
import { _ as _23, _ as _24 } from "../../BFormInput.vue_vue_type_script_setup_true_lang-B_LSzH-b.mjs";
import { _ as _25, a as a12 } from "../../BFormRadioGroup.vue_vue_type_script_setup_true_lang-DN3wYGro.mjs";
import { _ as _26, a as a13 } from "../../BFormSelect.vue_vue_type_script_setup_true_lang-DGM5cO4V.mjs";
import { _ as _27 } from "../../useFormSelect-CPaZDkF4.mjs";
import { _ as _28 } from "../../BFormSpinbutton.vue_vue_type_script_setup_true_lang-GGDbnpF-.mjs";
import { a as a14, _ as _29 } from "../../BFormTags.vue_vue_type_script_setup_true_lang-DPQdv7nh.mjs";
import { _ as _30 } from "../../BFormTextarea.vue_vue_type_script_setup_true_lang-DuQL-FCJ.mjs";
import { _ as _31 } from "../../BImg.vue_vue_type_script_setup_true_lang-B4G4OXC_.mjs";
import { _ as _32, a as a15 } from "../../BInputGroupText.vue_vue_type_script_setup_true_lang-xsphPcXn.mjs";
import { _ as _33 } from "../../BLink.vue_vue_type_script_setup_true_lang-DS4JR4ch.mjs";
import { _ as _34, a as a16 } from "../../BListGroupItem.vue_vue_type_script_setup_true_lang-DjbTQWnl.mjs";
import { B as B2, _ as _35 } from "../../BModalOrchestrator.vue_vue_type_script_setup_true_lang-DESSsMw4.mjs";
import { _ as _36, a as a17, b as b4, c as c4, d as d3 } from "../../BNavText.vue_vue_type_script_setup_true_lang-DD9a9EtK.mjs";
import { _ as _37, a as a18, b as b5, c as c5 } from "../../BNavbarToggle.vue_vue_type_script_setup_true_lang-J6tZn2vc.mjs";
import { _ as _38 } from "../../BOffcanvas.vue_vue_type_script_setup_true_lang-CGabsDn7.mjs";
import { _ as _39 } from "../../BOverlay.vue_vue_type_script_setup_true_lang-lCwuaPXO.mjs";
import { _ as _40 } from "../../BPagination.vue_vue_type_script_setup_true_lang-Dprt0p13.mjs";
import { _ as _41, a as a19, b as b6, c as c6, d as d4 } from "../../BPlaceholderWrapper.vue_vue_type_script_setup_true_lang-quMW6XOc.mjs";
import { _ as _42 } from "../../floatingUi-5pxbgyLd.mjs";
import { _ as _43 } from "../../BPopoverOrchestrator.vue_vue_type_script_setup_true_lang-CSpk5s_J.mjs";
import { _ as _44, a as a20 } from "../../BProgress.vue_vue_type_script_setup_true_lang-C-D_koFH.mjs";
import { _ as _45 } from "../../BSpinner.vue_vue_type_script_setup_true_lang-CXYbvCe5.mjs";
import { _ as _46, a as a21, b as b7, c as c7, d as d5, e as e3, f as f3, g } from "../../BTable.vue_vue_type_script_setup_true_lang-DSox5TGm.mjs";
import { _ as _47 } from "../../BTableSimple.vue_vue_type_script_setup_true_lang-Be44iN5l.mjs";
import { _ as _48, a as a22 } from "../../BTabs.vue_vue_type_script_setup_true_lang-Cb0FBpiV.mjs";
import { _ as _49, a as a23 } from "../../BToastOrchestrator.vue_vue_type_style_index_0_lang-CFmlSsxb.mjs";
import { _ as _50, a as a24 } from "../../BTooltipOrchestrator.vue_vue_type_script_setup_true_lang-ChH3484e.mjs";
export {
  _ as BAccordion,
  a as BAccordionItem,
  B as BAlert,
  _2 as BAvatar,
  a2 as BAvatarGroup,
  _3 as BBadge,
  _4 as BBreadcrumb,
  a3 as BBreadcrumbItem,
  _5 as BButton,
  _6 as BButtonGroup,
  a4 as BButtonToolbar,
  _8 as BCard,
  a5 as BCardBody,
  b as BCardFooter,
  _9 as BCardGroup,
  c as BCardHeader,
  d as BCardImg,
  e as BCardSubtitle,
  a6 as BCardText,
  f as BCardTitle,
  _10 as BCarousel,
  a7 as BCarouselSlide,
  _7 as BCloseButton,
  _12 as BCol,
  _11 as BCollapse,
  _13 as BContainer,
  _15 as BDropdown,
  _16 as BDropdownDivider,
  a8 as BDropdownForm,
  b2 as BDropdownGroup,
  c2 as BDropdownHeader,
  d2 as BDropdownItem,
  e2 as BDropdownItemButton,
  f2 as BDropdownText,
  _17 as BForm,
  _20 as BFormCheckbox,
  a11 as BFormCheckboxGroup,
  _18 as BFormDatalist,
  _21 as BFormFile,
  a9 as BFormFloatingLabel,
  _22 as BFormGroup,
  _23 as BFormInput,
  _19 as BFormInvalidFeedback,
  _25 as BFormRadio,
  a12 as BFormRadioGroup,
  a10 as BFormRow,
  _26 as BFormSelect,
  _27 as BFormSelectOption,
  a13 as BFormSelectOptionGroup,
  _28 as BFormSpinbutton,
  a14 as BFormTag,
  _29 as BFormTags,
  b3 as BFormText,
  _30 as BFormTextarea,
  c3 as BFormValidFeedback,
  _31 as BImg,
  _24 as BInput,
  _32 as BInputGroup,
  a15 as BInputGroupText,
  _33 as BLink,
  _34 as BListGroup,
  a16 as BListGroupItem,
  B2 as BModal,
  _35 as BModalOrchestrator,
  _36 as BNav,
  a17 as BNavForm,
  b4 as BNavItem,
  c4 as BNavItemDropdown,
  d3 as BNavText,
  _37 as BNavbar,
  a18 as BNavbarBrand,
  b5 as BNavbarNav,
  c5 as BNavbarToggle,
  _38 as BOffcanvas,
  _39 as BOverlay,
  _40 as BPagination,
  _41 as BPlaceholder,
  a19 as BPlaceholderButton,
  b6 as BPlaceholderCard,
  c6 as BPlaceholderTable,
  d4 as BPlaceholderWrapper,
  _42 as BPopover,
  _43 as BPopoverOrchestrator,
  _44 as BProgress,
  a20 as BProgressBar,
  _14 as BRow,
  _45 as BSpinner,
  _48 as BTab,
  _46 as BTable,
  a21 as BTableLite,
  _47 as BTableSimple,
  a22 as BTabs,
  b7 as BTbody,
  c7 as BTd,
  d5 as BTfoot,
  e3 as BTh,
  f3 as BThead,
  _49 as BToast,
  a23 as BToastOrchestrator,
  _50 as BTooltip,
  a24 as BTooltipOrchestrator,
  g as BTr
};
