// bootstrap-vue-next/dist/bootstrap-vue-next.mjs
import { i } from "./index-sKc4HTpY.mjs";
import { i as i2 } from "./index-CXpBEYSc.mjs";
import { i as i3 } from "./index-i-UFv70c.mjs";
import { i as i4 } from "./index-B7BE_6yh.mjs";
import { c as componentNames, a as componentsWithExternalPath, b as composableNames, d as composablesWithExternalPath, e as directiveNames, f as directivesWithExternalPath } from "./index-BOdW9Hjb.mjs";
import { B, i as i5 } from "./index-BOdW9Hjb.mjs";
import { i as i6 } from "./index-Dt02d7RV.mjs";
import { useBreadcrumb } from "./src/composables/useBreadcrumb/index.mjs";
import { useColorMode } from "./src/composables/useColorMode/index.mjs";
import { useModal } from "./src/composables/useModal/index.mjs";
import { useModalController } from "./src/composables/useModalController/index.mjs";
import { useScrollspy } from "./src/composables/useScrollspy/index.mjs";
import { useToast } from "./src/composables/useToast/index.mjs";
import { useTooltip } from "./src/composables/useTooltip/index.mjs";
import { usePopover } from "./src/composables/usePopover/index.mjs";
import { vBColorMode } from "./src/directives/BColorMode/index.mjs";
import { vBToggle, vBToggle as vBToggle2 } from "./src/directives/BToggle/index.mjs";
import { vBPopover } from "./src/directives/BPopover/index.mjs";
import { vBScrollspy } from "./src/directives/BScrollspy/index.mjs";
import { vBTooltip } from "./src/directives/BTooltip/index.mjs";
import { b, B as B2, a } from "./classes-IC0yVJlq.mjs";
import { _, a as a2 } from "./BAccordionItem.vue_vue_type_script_setup_true_lang-CIm9Esfp.mjs";
import { B as B3 } from "./BAlert-ChvVF808.mjs";
import { _ as _2, a as a3 } from "./BAvatarGroup.vue_vue_type_script_setup_true_lang-DEn76BWm.mjs";
import { _ as _3 } from "./BBadge.vue_vue_type_script_setup_true_lang-CYoUdtux.mjs";
import { _ as _4, a as a4 } from "./BBreadcrumb.vue_vue_type_script_setup_true_lang-CGIh5Dyn.mjs";
import { _ as _5 } from "./BButton.vue_vue_type_script_setup_true_lang-DUrERyIC.mjs";
import { _ as _6, a as a5 } from "./BButtonToolbar.vue_vue_type_script_setup_true_lang-DeG8adnx.mjs";
import { _ as _7 } from "./BCloseButton.vue_vue_type_script_setup_true_lang-B_GsCJs6.mjs";
import { _ as _8, a as a6, b as b2, c, d, e, f } from "./BCard.vue_vue_type_script_setup_true_lang-Bmyix7rF.mjs";
import { _ as _9, a as a7 } from "./BCardText.vue_vue_type_script_setup_true_lang-DHmRlQE7.mjs";
import { _ as _10, a as a8 } from "./BCarouselSlide.vue_vue_type_script_setup_true_lang-CGqYLLHu.mjs";
import { _ as _11 } from "./BCollapse.vue_vue_type_script_setup_true_lang-DMQdKnW0.mjs";
import { _ as _12 } from "./BCol.vue_vue_type_script_setup_true_lang-CpPd5cQ6.mjs";
import { _ as _13 } from "./BContainer.vue_vue_type_script_setup_true_lang-CRsJ73Pp.mjs";
import { _ as _14 } from "./BRow.vue_vue_type_script_setup_true_lang-1tXs9iie.mjs";
import { _ as _15 } from "./BDropdown.vue_vue_type_script_setup_true_lang-CCBMSTuN.mjs";
import { _ as _16, a as a9, b as b3, c as c2, d as d2, e as e2, f as f2 } from "./BDropdownText.vue_vue_type_script_setup_true_lang-ChLJg_WW.mjs";
import { _ as _17 } from "./BForm.vue_vue_type_script_setup_true_lang-BniLBRHQ.mjs";
import { _ as _18, a as a10 } from "./BFormFloatingLabel.vue_vue_type_script_setup_true_lang-B9qmgxL3.mjs";
import { _ as _19, a as a11, b as b4, c as c3 } from "./BFormValidFeedback.vue_vue_type_script_setup_true_lang-DMdYTYKn.mjs";
import { _ as _20, a as a12 } from "./BFormCheckboxGroup.vue_vue_type_script_setup_true_lang-BD7pZJij.mjs";
import { _ as _21 } from "./BFormFile.vue_vue_type_script_setup_true_lang-BEtcRM_N.mjs";
import { _ as _22 } from "./BFormGroup.vue_vue_type_script_lang-CfL_PZZX.mjs";
import { _ as _23, _ as _24 } from "./BFormInput.vue_vue_type_script_setup_true_lang-B_LSzH-b.mjs";
import { _ as _25, a as a13 } from "./BFormRadioGroup.vue_vue_type_script_setup_true_lang-DN3wYGro.mjs";
import { _ as _26, a as a14 } from "./BFormSelect.vue_vue_type_script_setup_true_lang-DGM5cO4V.mjs";
import { _ as _27 } from "./useFormSelect-CPaZDkF4.mjs";
import { _ as _28 } from "./BFormSpinbutton.vue_vue_type_script_setup_true_lang-GGDbnpF-.mjs";
import { a as a15, _ as _29 } from "./BFormTags.vue_vue_type_script_setup_true_lang-DPQdv7nh.mjs";
import { _ as _30 } from "./BFormTextarea.vue_vue_type_script_setup_true_lang-DuQL-FCJ.mjs";
import { _ as _31 } from "./BImg.vue_vue_type_script_setup_true_lang-B4G4OXC_.mjs";
import { _ as _32, a as a16 } from "./BInputGroupText.vue_vue_type_script_setup_true_lang-xsphPcXn.mjs";
import { _ as _33 } from "./BLink.vue_vue_type_script_setup_true_lang-DS4JR4ch.mjs";
import { _ as _34, a as a17 } from "./BListGroupItem.vue_vue_type_script_setup_true_lang-DjbTQWnl.mjs";
import { B as B4, _ as _35 } from "./BModalOrchestrator.vue_vue_type_script_setup_true_lang-DESSsMw4.mjs";
import { _ as _36, a as a18, b as b5, c as c4, d as d3 } from "./BNavText.vue_vue_type_script_setup_true_lang-DD9a9EtK.mjs";
import { _ as _37, a as a19, b as b6, c as c5 } from "./BNavbarToggle.vue_vue_type_script_setup_true_lang-J6tZn2vc.mjs";
import { _ as _38 } from "./BOffcanvas.vue_vue_type_script_setup_true_lang-CGabsDn7.mjs";
import { _ as _39 } from "./BOverlay.vue_vue_type_script_setup_true_lang-lCwuaPXO.mjs";
import { _ as _40 } from "./BPagination.vue_vue_type_script_setup_true_lang-Dprt0p13.mjs";
import { _ as _41, a as a20, b as b7, c as c6, d as d4 } from "./BPlaceholderWrapper.vue_vue_type_script_setup_true_lang-quMW6XOc.mjs";
import { _ as _42 } from "./floatingUi-5pxbgyLd.mjs";
import { _ as _43 } from "./BPopoverOrchestrator.vue_vue_type_script_setup_true_lang-CSpk5s_J.mjs";
import { _ as _44, a as a21 } from "./BProgress.vue_vue_type_script_setup_true_lang-C-D_koFH.mjs";
import { _ as _45 } from "./BSpinner.vue_vue_type_script_setup_true_lang-CXYbvCe5.mjs";
import { _ as _46, a as a22, b as b8, c as c7, d as d5, e as e3, f as f3, g } from "./BTable.vue_vue_type_script_setup_true_lang-DSox5TGm.mjs";
import { _ as _47 } from "./BTableSimple.vue_vue_type_script_setup_true_lang-Be44iN5l.mjs";
import { _ as _48, a as a23 } from "./BTabs.vue_vue_type_script_setup_true_lang-Cb0FBpiV.mjs";
import { _ as _49, a as a24 } from "./BToastOrchestrator.vue_vue_type_style_index_0_lang-CFmlSsxb.mjs";
import { _ as _50, a as a25 } from "./BTooltipOrchestrator.vue_vue_type_script_setup_true_lang-ChH3484e.mjs";
import { breadcrumbPlugin } from "./src/plugins/breadcrumb/index.mjs";
import { createBootstrap } from "./src/plugins/createBootstrap/index.mjs";
import { defaultsPlugin } from "./src/plugins/defaults/index.mjs";
import { modalControllerPlugin } from "./src/plugins/modalController/index.mjs";
import { modalManagerPlugin } from "./src/plugins/modalManager/index.mjs";
import { popoverPlugin } from "./src/plugins/popover/index.mjs";
import { rtlPlugin } from "./src/plugins/rtl/index.mjs";
import { toastPlugin } from "./src/plugins/toast/index.mjs";
import { tooltipPlugin } from "./src/plugins/tooltip/index.mjs";
const index = /* @__PURE__ */ Object.freeze(/* @__PURE__ */ Object.defineProperty({
  __proto__: null,
  componentNames,
  componentsWithExternalPath,
  composableNames,
  composablesWithExternalPath,
  directiveNames,
  directivesWithExternalPath
}, Symbol.toStringTag, { value: "Module" }));
export {
  _ as BAccordion,
  a2 as BAccordionItem,
  B3 as BAlert,
  _2 as BAvatar,
  a3 as BAvatarGroup,
  _3 as BBadge,
  _4 as BBreadcrumb,
  a4 as BBreadcrumbItem,
  _5 as BButton,
  _6 as BButtonGroup,
  a5 as BButtonToolbar,
  _8 as BCard,
  a6 as BCardBody,
  b2 as BCardFooter,
  _9 as BCardGroup,
  c as BCardHeader,
  d as BCardImg,
  e as BCardSubtitle,
  a7 as BCardText,
  f as BCardTitle,
  _10 as BCarousel,
  a8 as BCarouselSlide,
  _7 as BCloseButton,
  _12 as BCol,
  _11 as BCollapse,
  _13 as BContainer,
  _15 as BDropdown,
  _16 as BDropdownDivider,
  a9 as BDropdownForm,
  b3 as BDropdownGroup,
  c2 as BDropdownHeader,
  d2 as BDropdownItem,
  e2 as BDropdownItemButton,
  f2 as BDropdownText,
  _17 as BForm,
  _20 as BFormCheckbox,
  a12 as BFormCheckboxGroup,
  _18 as BFormDatalist,
  _21 as BFormFile,
  a10 as BFormFloatingLabel,
  _22 as BFormGroup,
  _23 as BFormInput,
  _19 as BFormInvalidFeedback,
  _25 as BFormRadio,
  a13 as BFormRadioGroup,
  a11 as BFormRow,
  _26 as BFormSelect,
  _27 as BFormSelectOption,
  a14 as BFormSelectOptionGroup,
  _28 as BFormSpinbutton,
  a15 as BFormTag,
  _29 as BFormTags,
  b4 as BFormText,
  _30 as BFormTextarea,
  c3 as BFormValidFeedback,
  _31 as BImg,
  _24 as BInput,
  _32 as BInputGroup,
  a16 as BInputGroupText,
  _33 as BLink,
  _34 as BListGroup,
  a17 as BListGroupItem,
  B4 as BModal,
  _35 as BModalOrchestrator,
  _36 as BNav,
  a18 as BNavForm,
  b5 as BNavItem,
  c4 as BNavItemDropdown,
  d3 as BNavText,
  _37 as BNavbar,
  a19 as BNavbarBrand,
  b6 as BNavbarNav,
  c5 as BNavbarToggle,
  _38 as BOffcanvas,
  _39 as BOverlay,
  _40 as BPagination,
  _41 as BPlaceholder,
  a20 as BPlaceholderButton,
  b7 as BPlaceholderCard,
  c6 as BPlaceholderTable,
  d4 as BPlaceholderWrapper,
  _42 as BPopover,
  _43 as BPopoverOrchestrator,
  _44 as BProgress,
  a21 as BProgressBar,
  _14 as BRow,
  _45 as BSpinner,
  _48 as BTab,
  _46 as BTable,
  a22 as BTableLite,
  _47 as BTableSimple,
  a23 as BTabs,
  b8 as BTbody,
  c7 as BTd,
  d5 as BTfoot,
  e3 as BTh,
  f3 as BThead,
  _49 as BToast,
  a24 as BToastOrchestrator,
  _50 as BTooltip,
  a25 as BTooltipOrchestrator,
  g as BTr,
  B as BootstrapVueNextResolver,
  b as BvCarouselEvent,
  B2 as BvEvent,
  a as BvTriggerableEvent,
  i4 as Components,
  i as Composables,
  i2 as Directives,
  i6 as Plugins,
  i5 as Resolvers,
  index as Types,
  i3 as Utils,
  breadcrumbPlugin,
  componentNames,
  componentsWithExternalPath,
  composableNames,
  composablesWithExternalPath,
  createBootstrap,
  defaultsPlugin,
  directiveNames,
  directivesWithExternalPath,
  modalControllerPlugin,
  modalManagerPlugin,
  popoverPlugin,
  rtlPlugin,
  toastPlugin,
  tooltipPlugin,
  useBreadcrumb,
  useColorMode,
  useModal,
  useModalController,
  usePopover,
  useScrollspy,
  useToast,
  useTooltip,
  vBColorMode,
  vBToggle as vBModal,
  vBPopover,
  vBScrollspy,
  vBToggle2 as vBToggle,
  vBTooltip
};
